internal/provider/repository_branch_protection_resource.go

```diff
@@ -0,0 +1,862 @@
+	opts := r.modelToEditOption(ctx, &data)
+
+	tflog.Debug(ctx, "Update branch protection request parameters", map[string]any{
+		"options": opts,
+	})
+
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update branch protection
+	protection, res, err := r.client.EditBranchProtection(
+		repo.Owner.UserName,
+		repo.Name,
+		data.BranchName.ValueString(),
+		opts,
+	)
+	if err != nil {
+		var msg string
+		if res != nil {
+			tflog.Error(ctx, "Error", map[string]any{
+				"status": res.Status,
+			})
+
+			switch res.StatusCode {
+			case 403:
+				msg = fmt.Sprintf("Forbidden: %s", err)
+			case 404:
+				msg = fmt.Sprintf("Branch protection not found: %s", err)
+			case 422:
+				msg = fmt.Sprintf("Validation error: %s", err)
+			default:
+				msg = fmt.Sprintf("Unknown error: %s", err)
+			}
+		} else {
+			msg = fmt.Sprintf("Failed to update branch protection: %s", err)
+		}
+		resp.Diagnostics.AddError("Unable to update branch protection", msg)
+		return
+	}
+
+	// Update model with response data
+	diags = r.mapResponseToModel(protection, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update model with response data
+	diags = resp.State.Set(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *repositoryBranchProtectionResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	defer un(trace(ctx, "Delete repository branch protection resource"))
+
+	var data repositoryBranchProtectionResourceModel
+
+	diags := req.State.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	repositoryID := data.RepositoryId.ValueInt64()
+
+	repo, diags := r.fetchRepository(repositoryID)
+
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Info(ctx, "Delete repository branch protection", map[string]any{
+		"repository_id":    repositoryID,
+		"repository_name":  repo.Name,
+		"repository_owner": repo.Owner.UserName,
+		"branch_name":      data.BranchName.ValueString(),
+	})
+
+	// Delete branch protection
+	res, err := r.client.DeleteBranchProtection(
+		repo.Owner.UserName,
+		repo.Name,
+		data.BranchName.ValueString(),
+	)
+	if err != nil {
+		if res != nil && res.StatusCode == 404 {
+			// Already deleted
+			return
+		}
+
+		resp.Diagnostics.AddError(
+			"Unable to delete branch protection",
+			fmt.Sprintf("Error: %s", err),
+		)
+		return
+	}
+}
+
+func (r *repositoryBranchProtectionResource) ImportState(ctx context.Context, request resource.ImportStateRequest, response *resource.ImportStateResponse) {
+	defer un(trace(ctx, "Import repository branch protection resource"))
+
+	parts := strings.Split(request.ID, "/")
+	if len(parts) != 3 {
+		response.Diagnostics.AddError(
+			"Invalid Import ID",
+			fmt.Sprintf("Expected import ID in format 'owner/repo/branch_name', got: %s", request.ID),
```

Comment:
Please add the import identifier syntax (`owner/repo/branch`) to the table in the documentation: https://github.com/svalabs/terraform-provider-forgejo/blob/07b595e6a829e109fb29849ed09f0cf0814c4880/README.md?plain=1#L183